rest/main.go

```diff
@@ -170,9 +171,31 @@ func serverMainPersistentConfig(fs *flag.FlagSet, flagStartupConfig *StartupConf
 		return false, err
 	}
 
+	initialStartupConfig, err := getInitialStartupConfig(fileStartupConfig, flagStartupConfig)
+	if err != nil {
+		return false, err
+	}
+
+	ctx.initialStartupConfig = initialStartupConfig
+
 	return false, startServer(&sc, ctx)
 }
 
+func getInitialStartupConfig(sc *StartupConfig, flagStartupConfig *StartupConfig) (*StartupConfig, error) {
+	initialStartupConfig := StartupConfig{}
+	err := initialStartupConfig.Merge(sc)
+	if err != nil {
+		return nil, err
+	}
+
+	err = initialStartupConfig.Merge(flagStartupConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &initialStartupConfig, err
```

Comment:
Am nitpicking but I'll commit so it's ready to merge tomorrow morning

```suggestion
	return &initialStartupConfig, nil
```